packages/stockflux-core/src/services/ChildWindowLauncher.js

```diff
@@ -1,18 +1,44 @@
 import * as OpenfinApiHelpers from '../openfin-api-utils/openfinApiHelpers';
 
+const windowAlreadyExist = async (windowName, childWindows) => {
+  for (let i = 0; i < childWindows.length; i++) {
+    const childWindowOptions = await childWindows[i].getOptions();
+    if (childWindowOptions.name === windowName) {
+      return true;
+    }
+  }
+  return false;
+};
+
 const createChildWindow = async options => {
   const childWindows = await OpenfinApiHelpers.getChildWindows();
   const currentOptions = await OpenfinApiHelpers.getCurrentWindowOptions();
 
   options.uuid = currentOptions.uuid;
-  for (let i = 0; i < childWindows.length; i++) {
-    const childWindowOptions = await childWindows[i].getOptions();
-    if (childWindowOptions.name === options.name) {
-      if (childWindows[i]) {
-        childWindows[i].bringToFront();
-        return true;
+
+  if (options.allowMultiple) {
+    if (await windowAlreadyExist(options.name, childWindows)) {
+      let windowNumber = 1;
+      while (
```

Comment:
Maybe I'm missing something, but could you not filter a set of `childWindows` for ones of the same name and then just get the length and use that as the index of the next window?

Reply:
My theory was to re-use number as the become available. for example when you first open a window the list of windows would be 
`[name]`
opening a second would be
`[name,name1]`
removing the first window, it would end up 
`[name1]`
If I used the length of the list to work out the last index, in this instance it would try and make `name1` again. The way used above it would use `name` and then if a 3rd window was added `name2` would be used.

Reply:
Fair point, I hadn't considered that windows could be removed!